File: AppGenerateFiles/dbo/FrontModel/Catalogo_Cuentas.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools, BasicStates } from "../../WDevCore/WModules/WComponentsTools.js";
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { Catalogo_Sucursales }  from './Catalogo_Sucursales.js'
import { Catalogo_Sucursales }  from './Catalogo_Sucursales.js'
import { Categoria_Cuentas }  from './Categoria_Cuentas.js'
import { Detail_Movimiento }  from './Detail_Movimiento.js'
import { Transaction_Ingresos_Egresos }  from './Transaction_Ingresos_Egresos.js'
class Catalogo_Cuentas extends EntityClass {
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ id_cuentas;
   /**@type {String}*/ nombre;
   /**@type {String}*/ tipo_cuenta;
   /**@type {Number}*/ saldo;
   /**@type {Number}*/ saldo_dolares;
   /**@type {Boolean}*/ permite_dolares;
   /**@type {Boolean}*/ permite_cordobas;
   /**@type {Catalogo_Sucursales} ManyToOne*/ Catalogo_Sucursales;
   /**@type {Catalogo_Sucursales} ManyToOne*/ Catalogo_Sucursales;
   /**@type {Categoria_Cuentas} ManyToOne*/ Categoria_Cuentas;
   /**@type {Array<Detail_Movimiento>} OneToMany*/ Detail_Movimiento;
   /**@type {Array<Transaction_Ingresos_Egresos>} OneToMany*/ Transaction_Ingresos_Egresos;
}
export { Catalogo_Cuentas }

File: AppGenerateFiles/dbo/Model/Catalogo_Cuentas.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Catalogo_Cuentas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? id_cuentas { get; set; }
       public string? nombre { get; set; }
       public string? tipo_cuenta { get; set; }
       public int? id_sucursal { get; set; }
       public int? id_categoria { get; set; }
       public Double? saldo { get; set; }
       public Double? saldo_dolares { get; set; }
       public bool? permite_dolares { get; set; }
       public bool? permite_cordobas { get; set; }
       [ManyToOne(TableName = "Catalogo_Sucursales", KeyColumn = "Id_Sucursal", ForeignKeyColumn = "id_sucursal")]
       public Catalogo_Sucursales? Catalogo_Sucursales_id_sucursal { get; set; }
       [ManyToOne(TableName = "Catalogo_Sucursales", KeyColumn = "Id_Sucursal", ForeignKeyColumn = "id_sucursal")]
       public Catalogo_Sucursales? Catalogo_Sucursales_id_sucursal { get; set; }
       [ManyToOne(TableName = "Categoria_Cuentas", KeyColumn = "id_categoria", ForeignKeyColumn = "id_categoria")]
       public Categoria_Cuentas? Categoria_Cuentas { get; set; }
       [OneToMany(TableName = "Detail_Movimiento", KeyColumn = "id_cuentas", ForeignKeyColumn = "id_cuenta")]
       public List<Detail_Movimiento>? Detail_Movimiento { get; set; }
       [OneToMany(TableName = "Transaction_Ingresos_Egresos", KeyColumn = "id_cuentas", ForeignKeyColumn = "id_cuenta")]
       public List<Transaction_Ingresos_Egresos>? Transaction_Ingresos_Egresos { get; set; }
   }
}
